.github/dependabot.yml

```
@@ -1,11 +1,26 @@
 ---
 version: 2
+enable-beta-ecosystems: true
 updates:
   - package-ecosystem: "gomod"
     directory: "/"
     schedule:
       interval: "weekly"
     rebase-strategy: "disabled"
+    groups:
+      golang-dependencies:
+        patterns:
+          - "github.com/golang*"
+      k8s-dependencies:
+        patterns:
+          - "k8s.io*"
+          - "sigs.k8s.io*"
+      github-dependencies:
+        patterns:
+          - "github.com*"
```

```suggestion
          - "k8s.io*"
          - "sigs.k8s.io*"
      github-dependencies:
        patterns:
          - "github.com*"
```

I think for `k8s.io*"` and `sigs.k8s.io*"`, singe `*` make sense since it has only once string to match for example 

```
k8s.io/cloud-provider
```
only `cloud-provider`

but for `github.com` I think it will `github.com/**` since it has to make a recursive search
```
github.com/spf13/cobra
```
`/spf13` and `/cobra`


I guess I'm not confusing things but I have seen `/**` for recursive search for example branch search 
```
tags:
      - v*
    branches:
      - master
      - release-*
  pull_request:
    branches:
      - master
      - release-*
    paths-ignore:
      - "Documentation/**"
 ```
from here https://github.com/rook/rook/blob/master/.github/workflows/canary-integration-test.yml#L4-L15
      - "design/**"

---

Thanks @subhamkrai For me this looks to be enough, will revisit this if required in the followup PR.